### ISSUE TITLE
Support for knowledge base

### ISSUE THREAD

**Issue by jfaraklit** [Shaunwei/RealChar#129]
Support for a private knowledge base. Like Azure cognitive search + OpenAI Service.

**Comment by Shaunwei** [comment]
It's on our roadmap, stay tuned

**Comment by nuocamp** [comment]
so good

**Comment by scooter7** [comment]
Hi, related to this, if I copied a bunch of website text into the character profile, would that work in terms of enabling a user to ask questions about the content and have the bot respond with the information in the character profile? So, basically, I'm wanting to create a Q&A bot that can provide specific information. Thanks!

**Comment by jfaraklit** [comment]
@scooter7  yes. make a new character of your choice and add your data in the data folder. you might need to install some py libraries. for example for pdf pypdf and etc.

**Comment by johndpope** [comment]
https://github.com/StanGirard/quivr

**Comment by Kevin-free** [comment]
👍You guys are amazing! A knowledge base that supports uploading external files has been implemented.

But I also want to make a request to improve the product and users:
**Can a knowledge base be built from data generated by the current system (for example, chat records)?**

Or the website (for example, Twitter, Weibo). I saw `quivr` has this function, and @Shaunwei is a contributor, so it should be familiar.
